src/main/java/core/basesyntax/transaction/Transaction.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax.transaction;
+
+public class Transaction {
+    private static final String BALANCE_ABBREVIATION = "b";
+    private static final String SUPPLY_ABBREVIATION = "s";
+    private static final String PURCHASE_ABBREVIATION = "p";
+    private static final String RETURN_ABBREVIATION = "r";
+    private Operation operation;
+    private String product;
+    private int quantity;
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public String getProduct() {
+        return product;
+    }
+
+    public void setProduct(String fruit) {
+        this.product = fruit;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation getByCode(String code) {
+            for (Operation operationName : Operation.values()) {
+                if (operationName.getCode().equals(code)) {
+                    return operationName;
+                }
+            }
+            throw new RuntimeException("Invalid operation code");

```

Comment:
```suggestion
            throw new IllegalArgumentException("Invalid operation code " + code);
```
Always try to create as specific exceptions as possible, and include all the relevant details, to make debugging later on easier